runtime/gc_base/RootScanner.cpp

```diff
@@ -536,7 +537,23 @@ MM_RootScanner::scanOneThread(MM_EnvironmentBase *env, J9VMThread* walkThread, v
 		doVMThreadSlot(slot, &vmThreadIterator);
 	}
 
-	GC_VMThreadStackSlotIterator::scanSlots((J9VMThread *)env->getOmrVMThread()->_language_vmthread, walkThread, localData, stackSlotIterator, isStackFrameClassWalkNeeded(), _trackVisibleStackFrameDepth);
+	J9VMThread *currentThread = (J9VMThread *)env->getOmrVMThread()->_language_vmthread;
+	GC_VMThreadStackSlotIterator::scanSlots(currentThread, walkThread, localData, stackSlotIterator, isStackFrameClassWalkNeeded(), _trackVisibleStackFrameDepth);
+
+#if JAVA_SPEC_VERSION >= 19
+	if (NULL != walkThread->currentContinuation)
+	{
+		bool const compressed = _extensions->compressObjectReferences();
+		j9object_t carrierThreadObject = walkThread->carrierThreadObject;
+		MM_ForwardedHeader forwardHeader(carrierThreadObject, compressed);
+		if (forwardHeader.isForwardedPointer()) {
+			carrierThreadObject = forwardHeader.getForwardedObject();
+		}
```

Comment:
You should put a comment, something like this:
At this point we know that a virtual thread is mounted. We previously scanned its stack, and now we need to scan carrier's stack,too. Currently, Continuation struct is pointing to it. doSlot will now only push Continuation object for scanning, what will eventually get Continuation object and struct scanned, but carrier's stack too.